### PR TITLE
[add] nin API authentication

### PR DESCRIPTION
**Pull request by xand6r**
### What does this PR do?
- This PR initializes the nodejs file structure and a firebase function API

#### Description of Task to be completed?
- Add an API which is used to authenticate NINs and phone numbers 

#### Checklist:
- [x] Add NIN authentication API
- [x] Add express js and cors to the project 
- [x] configure express to work with json post requests

### CODE REVIEW

File: functions/routes/authentication.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+var express=require("express");
+var router=express.Router();
+
+// a route which we will be using to authenticate API calls for NIN pending our approval, it requires a post request to be activated
+router.post("/",(request,response)=>{
+    // extract the phone number and NIN from the body of the response
+    const {nin,phoneNumber}=request.body;
```
Comment (Way2nnadi) [inline functions/routes/authentication.js:7]: what is `nin`?

**Comment by Way2nnadi** [comment]
@Xand6r Code looks good, but have a ton of linting errors. make sure that this repo has the eslint rules from the `project-template` and that your code is meeting them

File: functions/routes/authentication.js (hunk 1 above)
Comment (xand6r) [reply functions/routes/authentication.js:7]: National identification number, there is a government API which can verify the NINs and phone numbers attaxhed

Comment (xand6r) [reply functions/routes/authentication.js:7]: It's a unique number given as our national identity.

Comment (Way2nnadi) [reply functions/routes/authentication.js:7]: okay, fix all the linting errors then you are all set to merge this code down

**Comment by Onyimatics** [comment]
[Create an integration with firebase phone number authentication](https://trello.com/c/1HwIr6IV/43-create-an-integration-with-firebase-phone-number-authentication)

**Review by Onyimatics — CHANGES_REQUESTED** [review]
Good job  @Xand6r. Looks good to me. However, there's conflict on this branch and needs to be resolved. Thank you

**Review by Onyimatics — APPROVED** [review]

**Review by usmanoa — APPROVED** [review]
Nice work

**Pull request merged**